subsys/mgmt/mcumgr/grp/img_mgmt/src/img_mgmt_state.c

```diff
@@ -518,6 +596,53 @@ int img_mgmt_set_next_boot_slot(int slot, bool confirm)
 
 	return rc;
 }
+#else
+int img_mgmt_set_next_boot_slot(int slot, bool confirm)
+{
+	const struct flash_area *fa;
+	int area_id = img_mgmt_flash_area_id(slot);
+	int rc = 0;
+	int active_image = img_mgmt_active_image();
+	int active_slot = img_mgmt_active_slot(active_image);
+
+	if (flash_area_open(area_id, &fa) != 0) {
+		return IMG_MGMT_RET_RC_FLASH_OPEN_FAILED;
+	}
+
+	rc = boot_set_next(fa, slot == active_slot, confirm);
+	if (rc != 0) {
+		/* Failed to set next slot for boot as desired */
+		if (slot == active_slot) {
+			LOG_ERR("Failed to write confirmed flag: %d", rc);
+		} else {
+			LOG_ERR("Failed to write pending flag for slot %d: %d", slot, rc);
+		}
+
+		/* Translate from boot util error code to IMG mgmt group error code */
+		if (rc == BOOT_EFLASH) {
+			rc = IMG_MGMT_RET_RC_FLASH_WRITE_FAILED;
+		} else if (rc == BOOT_EBADVECT) {
+			rc = IMG_MGMT_RET_RC_INVALID_IMAGE_VECTOR_TABLE;
+		} else if (rc == BOOT_EBADIMAGE) {
+			rc = IMG_MGMT_RET_RC_INVALID_IMAGE_HEADER_MAGIC;
+		} else {
+			rc = IMG_MGMT_RET_RC_UNKNOWN;
+		}
+	}
+	flash_area_close(fa);
+
+#if defined(CONFIG_MCUMGR_GRP_IMG_STATUS_HOOKS)
+	if (slot == active_slot && confirm) {
+		uint16_t ret_group;
+		/* Confirm event is only sent for active slot */
+		(void)mgmt_callback_notify(MGMT_EVT_OP_IMG_MGMT_DFU_CONFIRMED, NULL, 0, &ret_rc,
```

Comment:
`ret_rc` seems not declared here. This must be aligned to prevent build failures

Reply:
Aligned locally